pyramid/config/__init__.py

```diff
@@ -903,6 +921,13 @@ def end(self):
         """
         return self.manager.pop()
 
+    def __enter__(self):
+        self.begin()
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        self.end()
```

Comment:
I think it could be done similar to how we handle ``config.route_prefix`` and a few other state variables where we shuffle them through each new configurator. We could actually use that state to warn a user if they try to do a manual commit as well.